dipy/reconst/dti.py

```diff
@@ -1313,25 +1378,43 @@ def wrapped_fit_tensor(design_matrix, data, return_S0_hat=False,
                                   return_S0_hat=return_S0_hat,
                                   *args, **kwargs)
             data = data.reshape(-1, data.shape[-1])
-            dtiparams = np.empty((size, 12), dtype=np.float64)
+            if 'return_lower_triangular' in kwargs:
+                sz = 7 if kwargs['return_lower_triangular'] else 12
+            else:
+                sz = 12
```

Comment:
this is a bit repetitive. `sz = 7 if kwargs.get('return_lower_triangular') else 12` should be enough

Reply:
What if `return_lower_triangular` is in the dictionary but its value is `False` ?